### PR TITLE
Accessibility more fixes

### PR DESCRIPTION
**Pull request by jlhogan**
## Changes

Accessibility needs from screen reader walk through

Fixes https://app.asana.com/0/0/1203261751786617/f, https://app.asana.com/0/0/1203261751786618/f, https://app.asana.com/0/0/1203228691996331/f

### CODE REVIEW

File: e2e/search.spec.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ test("Clearable checkbox filter works on desktop", async ({ page }) => {
   await expect(activeFilters).toContainText("Tuesday");
 
   const clearAll = page.locator("#active-filters-section button", {
-    hasText: "Clear",
+    hasText: "Clear Tuesday",
```
Comment (cathydeng) [inline e2e/search.spec.ts:136]: i think we actually want this to test the "Clear all filters" button

Comment (jlhogan) [reply e2e/search.spec.ts:136]: WHOOPS yes

**Review by cathydeng — APPROVED** [review]
🙏

**Review by jessethach — APPROVED** [review]
LGTM!

**Pull request merged**